src/ui/public/kbn_top_nav/kbn_top_nav.js

```diff
@@ -74,10 +70,23 @@ module.directive('kbnTopNav', function (Private) {
         </div>
       `;
     },
-    controller($scope, $compile, $attrs, $element) {
+    controller($scope, $attrs, $element) {
       const KbnTopNavController = Private(KbnTopNavControllerProvider);
+      const navbarExtensions = Private(RegistryNavbarExtensionsProvider);
+      const getNavbarExtensions = _.memoize(function (name) {
```

Comment:
This should be moved outside of this controller function for the memoize to work correctly, no? Otherwise a new instance of these is made every time you use this directive.